feat: add omni_exportBundlerPrivateKey RPC method

silva-fj opened the pull request:
Adds a new RPC method `omni_exportBundlerPrivateKey` to export the bundler private key (accounting ECDSA signer key) for internal use.

## Implementation
- Protected by ECDSA signature verification with timestamp-based replay attack prevention
- Uses `OE_BUNDLER_KEY_EXPORT_AUTHORIZED_PUBKEY` env var to configure authorized public key
- Timestamp must be within 5-minute window and cannot be reused
- Client provides RSA-encrypted AES key in request parameters
- Server decrypts AES key using shielding key's RSA private key
- Bundler private key is encrypted with AES before returning as `SerdeAesOutput`

## Security
- Two-factor security: ECDSA signature authentication + RSA-encrypted AES transport
- Bundler private key never transmitted in plain text
- Signature verification ensures only authorized key holder can call method
- Timestamp validation prevents replay attacks via persistent storage
- Future timestamp check prevents clock manipulation

silva-fj committed:
feat: add bundler key export authorized pubkey config

silva-fj committed:
feat: add bundler key params to RpcContext

silva-fj committed:
feat: parse and pass bundler key params to RPC server

silva-fj committed:
feat: add omni_exportBundlerPrivateKey RPC method with tests

BillyWooo reviewed (commented):
Hi @silva-fj , thanks. It looks quite good for me. 
Besides, can you also check and update the ci yml files, dockerfiles, and gramine manifest file?  Just search around `WILDMETA_BACKEND_ECDSA_PUBKEY` for reference.

silva-fj committed:
feat: add bundler key export env var to config files

Add OE_BUNDLER_KEY_EXPORT_AUTHORIZED_PUBKEY to Dockerfile, manifest template, and env files for consistency across deployment configurations.

silva-fj committed:
feat: encrypt bundler private key in export RPC response

- Add RSA-encrypted AES key parameter to request
- Decrypt AES key using shielding key
- Encrypt bundler private key with AES before returning
- Update tests to handle encrypted response
- Add TEST_AES_KEY constant for test consistency

silva-fj committed:
refactor(main): remove redundant comment in key parsing

silva-fj committed:
Merge branch 'dev' into p-1716-support-export-bundler-private-key

silva-fj committed:
fix(tests): add missing hex_encode import

silva-fj committed:
Merge branch 'dev' into p-1716-support-export-bundler-private-key

BillyWooo reviewed (commented):
Hi @silva-fj , thank you. The implementation is really good.

Besides, can you also add one example / or extend current subcommand to generate out the rpc call command? Similar like we did in Signer/Revvy:
1. user call omni_getShieldingKey
2. use the ShieldingKey to create the signature, create new RSA-public key, and  construct the full curl command and print it out
3. user can copy and execute the full curl command directly to get encrypted bundler private key
4. use RSA-private key to decrypt out bundler private key.

silva-fj commented:
> Hi @silva-fj , thank you. The implementation is really good.
> 
> Besides, can you also add one example / or extend current subcommand to generate out the rpc call command? Similar like we did in Signer/Revvy:
> 
> 1. user call omni_getShieldingKey
> 2. use the ShieldingKey to create the signature, create new RSA-public key, and  construct the full curl command and print it out
> 3. user can copy and execute the full curl command directly to get encrypted bundler private key
> 4. use RSA-private key to decrypt out bundler private key.

Yes, we can extend the cli to do that. How about if we have a separate issue for this?

Kailai-Wang reviewed (commented):
Thanks!

Yes - we would need a CLI to generate the export payload, given e.g. path to the private key file (similar to builder key import)

Additionally, I don't see `OE_BUNDLER_KEY_EXPORT_AUTHORIZED_PUBKEY` being written when building the release image - could you check it?

silva-fj committed:
fix(omni-executor): add missing env vars to release stage

Add OE_WILDMETA_BACKEND_ECDSA_PUBKEY and OE_BUNDLER_KEY_EXPORT_AUTHORIZED_PUBKEY
to the release Docker stage. These are required at runtime by the Gramine manifest
template but were only defined in the builder stage.

silva-fj committed:
Merge branch 'dev' into p-1716-support-export-bundler-private-key

silva-fj committed:
fix(omni-executor): remove parachain dependencies from export_bundler_private_key

Remove unused parentchain RPC client imports and generic parameters from
register_export_bundler_private_key function to align with RpcContext
refactoring that removed parachain connection.
